### PR TITLE
[REVIEW] Ensure no packages from conda defaults channel

### PR DESCRIPTION
**Pull request by dillon-cullinan**
This PR ensures no docker image build succeeds if it contains a package from the conda defaults channel.

Also adds some support in the `generate_dockerfiles.py` file to allow for global partials.

Logic has been tested and confirmed working on runtime containers.

### CODE REVIEW

**Comment by ajschmidt8** [comment]
This test is useful, but I'm not sure it belongs in the Dockerfile steps. I think it'd be more suitable to add to the tests that we run on the images once they're already built. Adding tests to a Dockerfile doesn't seem like a good practice to initiate.

File: templates/partials/ensure-no-defaults.dockerfile.j2 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{# This partial file is used to ensure there are no packages from the defaults conda channel #}
+
+{# Check for defaults packages in rapids env #}
+RUN source activate rapids \
```
Comment (raydouglass) [inline templates/partials/ensure-no-defaults.dockerfile.j2:4]: Should we output `conda list` before the check so if it fails we can see what packages were in `defaults`?

**Comment by raydouglass** [comment]
> Adding tests via Dockerfile steps doesn't seem like a good practice to initiate.

I think I agree with this.

File: generated-dockerfiles/rapidsai-clx_centos7-base.Dockerfile [hunk 2]
```diff
@@ -24,6 +24,10 @@ WORKDIR ${RAPIDS_DIR}
 
 RUN conda clean -afy
 
+
+RUN source activate rapids \
+    && CONDA_DEFAULT_PKG_NUM=`conda list | awk '{ print $4 }' | grep defaults | wc -l` \
```
Comment (mattf) [inline generated-dockerfiles/rapidsai-clx_centos7-base.Dockerfile:29]: fyi: grep defaults | wc -l -> grep -c defaults

File: templates/partials/ensure-no-defaults.dockerfile.j2 [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+    && CONDA_DEFAULT_PKG_NUM=`conda list | awk '{ print $4 }' | grep defaults | wc -l` \
```
Comment (mattf) [inline templates/partials/ensure-no-defaults.dockerfile.j2:5]: fyi: `grep defaults | wc -l` is shorter as `grep -c defaults`

File: templates/partials/ensure-no-defaults.dockerfile.j2 (hunk 1 above)
Comment (jakirkham) [inline templates/partials/ensure-no-defaults.dockerfile.j2:4]: Should we also check the `base` environment as well?

Comment (dillon-cullinan) [reply templates/partials/ensure-no-defaults.dockerfile.j2]: Ah, that makes sense.

Comment (jakirkham) [reply templates/partials/ensure-no-defaults.dockerfile.j2]: Right Conda's. Sorry for the confusion

**Pull request closed without merging**